app/models/endpoint.rb

```diff
@@ -41,6 +41,18 @@ def ssl_cert_store
     nil # use system defaults
   end
 
+  # Split concatenated PEM certs to a list
+  def certificate_authorities
+    return [] if certificate_authority.blank?
+    certificate_authority.split(/(?=-----BEGIN)/).reject(&:blank?)
```

Comment:
It is unlikely, but PEM files may contain `-----BEGIN` mark at positions other than the first column. This regular expression will match that. I'd suggest you add an anchor: `/^(=?-----BEGIN)/`.